github/resource_github_repository.go

```diff
@@ -309,11 +309,6 @@ func resourceGithubRepositoryCreate(d *schema.ResourceData, meta interface{}) er
 	}
 
 	repoReq.Private = github.Bool(isPrivate)
-	if isPrivate {
-		repoReq.Visibility = github.String("private")
-	} else {
-		repoReq.Visibility = github.String("public")
-	}
```

Comment:
This block was overriding the `Visibility` property which was already set on [line 262](https://github.com/integrations/terraform-provider-github/pull/794/files#diff-804ba2b1845dd42f7c9494886ff5a31bc1f7b1affea204b6b1b1077248f563c4R262) by the call to `resourceGithubRepositoryObject` on [line 287](https://github.com/integrations/terraform-provider-github/pull/794/files#diff-804ba2b1845dd42f7c9494886ff5a31bc1f7b1affea204b6b1b1077248f563c4R287)